Assets/Scripts/Systems/Modules/ModulesInfo.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+using Ships.Components;
+using Systems.Save;
+using UnityEngine;
+
+namespace Systems.Modules
+{
+    [Serializable]
+    public class ModulesInfo : MonoBehaviour, ISavable
+    {
+        public Module[,] Grid;
+        public List<Module> Modules;
+        public int RowHeight;
+        public int ColumnLength;
+
+        private ShipInfo _info;
+
+        private void Awake()
+        {
+            _info = GetComponent<ShipInfo>();
+            if (Grid == null)
+            {
+                Grid = ModuleList.ListsToGrid(_info.Data.ModuleGrid);
+            }
+
+            if (Modules == null)
+            {
+                Modules = new List<Module>();
+            }
+        }
+
+        public void AddModule(Module module, int row, int column)
+        {
+            foreach(Coordinates coords in module.Data.GridPositions)
+            {
+                if (column + coords.x >= 0 && column + coords.x < ColumnLength &&
+                    row + coords.y >= 0 && row + coords.y < RowHeight)
+                {
+                    Grid[row + coords.y, column + coords.x] = module;
+                }
+            }
```

Comment:
I feel like there should be a check so you can't add modules in invalid positions